cmd/geth/dbcmd.go

```diff
@@ -537,41 +537,51 @@ func dbDumpTrie(ctx *cli.Context) error {
 
 func freezerInspect(ctx *cli.Context) error {
 	var (
-		start, end    int64
-		disableSnappy bool
-		err           error
+		start, end int64
+		err        error
 	)
 	if ctx.NArg() < 3 {
 		return fmt.Errorf("required arguments: %v", ctx.Command.ArgsUsage)
 	}
 	kind := ctx.Args().Get(0)
-	if noSnap, ok := rawdb.FreezerNoSnappy[kind]; !ok {
-		var options []string
-		for opt := range rawdb.FreezerNoSnappy {
-			options = append(options, opt)
-		}
-		sort.Strings(options)
-		return fmt.Errorf("Could read freezer-type '%v'. Available options: %v", kind, options)
-	} else {
-		disableSnappy = noSnap
+
+	exist, disableSnappy, subdir := rawdb.FreezerTableInfo(kind)
+	if !exist {
+		tables := rawdb.FreezerTables()
+		sort.Strings(tables)
+		return fmt.Errorf("could read freezer-type '%v'. Available options: %v", kind, tables)
```

Comment:
I know it's not your error, but might as well fix it now: 
```suggestion
		return fmt.Errorf("could not read freezer-type '%v'. Available options: %v", kind, tables)
```